src/middleware.js

```diff
@@ -160,7 +160,7 @@ function apiMiddleware({ getState }) {
         return next(
           await actionWith(
             {
-              ...requestType,
+              ...failureType,
```

Comment:
maybe a "migration" helper option could exist? like:
```
   ...(options.legacyApi ? requestType : failureType),
```
maybe until v4 or something, although not sure if anyone else might find this useful, can always stick with v2 till you're ready for migration too :)

Reply:
I'm open to a migration helper, but I don't know what the right approach is here. A breaking change that:

- requires you to go in and add a `legacyRequestError: true` to your `RSAA`s as a temporary fix

OR

- requires you to update your error handling paths to this new spec

I lean towards the latter but can be convinced otherwise!